src/replication/dao/registry.go

```diff
@@ -55,34 +49,37 @@ func GetRegistryByURL(url string) (*models.Registry, error) {
 	return &r, err
 }
 
-// ListRegistries lists registries. Registries returned are sorted by creation time.
-// - query: query to the registry name, name query and pagination are defined.
-func ListRegistries(query ...*ListRegistryQuery) (int64, []*models.Registry, error) {
-	o := dao.GetOrmer()
-
-	q := o.QueryTable(&models.Registry{})
-	if len(query) > 0 && len(query[0].Query) > 0 {
-		q = q.Filter("name__contains", query[0].Query)
+// ListRegistries lists registries
+func ListRegistries(ctx context.Context, query *q.Query) (int64, []*models.Registry, error) {
+	var countQuery *q.Query
+	if query != nil {
+		// ignore the page number and size
+		countQuery = &q.Query{
+			Keywords: query.Keywords,
+		}
 	}
-
-	total, err := q.Count()
+	countQs, err := liborm.QuerySetter(ctx, &models.Registry{}, countQuery)
```

Comment:
I may lack some background of the query but what's the different between countQs and qs? why not use the length of `registries` as the first return val?

Reply:
The `countQs` contains no pagination information. 
And the `registries` may contain only the subset of all registry records if the pagination is set in query